packages/protobuf/src/to-json.ts

```diff
@@ -135,19 +134,13 @@ function reflectToJson(msg: ReflectMessage, opts: JsonWriteOptions): JsonValue {
   const json: JsonObject = {};
   for (const f of msg.sortedFields) {
     if (!msg.isSet(f)) {
-      if (
-        f.fieldKind != "map" &&
-        f.fieldKind != "list" &&
-        f.presence == LEGACY_REQUIRED
-      ) {
+      if (f.presence == LEGACY_REQUIRED) {
         throw new Error(
           `cannot encode field ${msg.desc.typeName}.${f.name} to binary: required field not set`,
         );
       }
-      if (!opts.emitDefaultValues) {
-        continue;
-      }
-      if (!canEmitFieldDefaultValue(f)) {
+      if (!opts.emitDefaultValues || f.presence !== IMPLICIT) {
+        // Fields with implicit presence omit zero values (e.g. empty string) by default
```

Comment:
The function `canEmitFieldDefaultValue` is replaced by this simple check, matching the [specified behavior](https://protobuf.dev/programming-guides/proto3/#json-options):

> Fields that don’t support presence and that have their default value are omitted by default in JSON output (for example, an implicit presence integer with a 0 value, implicit presence string fields that are empty strings, and empty repeated and map fields). 

We might want to rename the serialization option `emitDefaultValues` to `emitFieldsWithoutPresence` as a follow-up.